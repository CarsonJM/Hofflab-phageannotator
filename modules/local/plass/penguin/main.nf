process PLASS_PENGUIN {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/plass:5.cf8933--pl5321h6a68c12_0':
        'biocontainers/plass:5.cf8933--pl5321h6a68c12_0' }"

    input:
    tuple val(meta), path(fastq)

    output:
    tuple val(meta), path("${prefix}.contigs.fasta.gz") , emit: contigs
    tuple val(meta), path("${prefix}.log")              , emit: log
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    if [ \$(zcat ${fastq[0]} | grep "@" | wc -l) -gt 0 ]; then
        penguin \\
            ${args} \\
            ${fastq} \\
            ${prefix}.contigs.fasta \\
            --threads ${task.cpus} \\
            --compressed 1 2> ${prefix}.log
    else
        echo "" | gzip > ${prefix}.contigs.fasta.gz
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        penguin: \$(echo \$(penguin 2>&1) | sed -n 's/^.*PenguiN Version: //; 3p' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.contigs.fasta.gz
    touch ${prefix}.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        penguin: \$(echo \$(penguin 2>&1) | sed -n 's/^.*PenguiN Version: //; 3p' ))
    END_VERSIONS
    """
}
