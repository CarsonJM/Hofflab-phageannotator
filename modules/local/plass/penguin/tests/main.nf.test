nextflow_process {

    name "Test Process: PLASS_PENGUIN"
    script "../main.nf"
    process "PLASS_PENGUIN"


    test("fastq.gz") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test1' ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }

    test("fastq.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test1' ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
