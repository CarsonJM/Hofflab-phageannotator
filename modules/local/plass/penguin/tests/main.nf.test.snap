{
    "fastq.gz": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "contigs": [
                    
                ],
                "log": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-08T16:49:37.581585688"
    }
}